api/_dexes/cross-swap.ts

```diff
@@ -242,23 +241,39 @@ export async function buildCrossSwapTx(
     toAddress = spokePool.address;
   }
 
-  const [gas, gasPrice] = await Promise.all([
-    spokePool.provider.estimateGas({
-      from: crossSwapQuotes.crossSwap.depositor,
-      ...tx,
-    }),
-    latestGasPriceCache(originChainId).get(),
-  ]);
-
   return {
     from: crossSwapQuotes.crossSwap.depositor,
     to: toAddress,
     data: integratorId ? tagIntegratorId(integratorId, tx.data!) : tx.data,
-    gas,
-    gasPrice,
     value: tx.value,
   };
 }
+async function extractDepositDataStruct(crossSwapQuotes: CrossSwapQuotes) {
+  const originChainId = crossSwapQuotes.crossSwap.inputToken.chainId;
+  const destinationChainId = crossSwapQuotes.crossSwap.outputToken.chainId;
+  const spokePool = getSpokePool(originChainId);
+  const message = crossSwapQuotes.bridgeQuote.message || "0x";
+  const deposit = {
+    depositor: crossSwapQuotes.crossSwap.depositor,
+    recipient:
+      message && message !== "0x"
+        ? getMultiCallHandlerAddress(destinationChainId)
+        : crossSwapQuotes.crossSwap.recipient,
```

Comment:
I don't think `message` can ever be undefined, and we have a utility function in the SDK that centralises the "message is empty" check:

```suggestion
    recipient: isMessageEmpty(message)
        ? crossSwapQuotes.crossSwap.recipient
        : getMultiCallHandlerAddress(destinationChainId),
```

Reply:
Also, is it safe to assume that if we have a non-empty message, then the recipient is the multicall handler, couldn't an integrator implement their own version of `HandleV3AcrossMessage`?

Reply:
@bmzig I think in general yes. But for the initial scope of this swap endpoint, we should be able to assume that API consumers won't use their own handler contract. Because we can not do the swapping for them then